src/component/graph.rs

```diff
@@ -299,399 +333,281 @@ impl Component {
+        return;
     }
 
-    for grid_line in &mut state.vertical_grid_lines {
-        *grid_line /= WIDTH;
+    let timing_method = timer.current_timing_method();
+    let mut live_delta = analysis::check_live_delta(timer, true, comparison, timing_method);
+    let current_time = timer.current_time()[timing_method];
+    let current_split_comparison = timer
+        .run()
+        .segment(draw_info.split_index)
+        .comparison(comparison)[timing_method];
+
+    if let (Some(current_time), Some(current_split_comparison), None) =
+        (current_time, current_split_comparison, live_delta)
+    {
+        // Live delta should be shown despite what analysis::check_live_delta says.
+        let delta = current_time - current_split_comparison;
+        if delta.total_seconds() as f32 > draw_info.min_delta {
+            live_delta = Some(delta);
+        }
     }
 
-    state.middle /= HEIGHT;
+    if let Some(live_delta) = live_delta {
+        let delta = live_delta.total_seconds() as f32;
+        if delta > draw_info.max_delta {
+            draw_info.max_delta = delta;
+        } else if delta < draw_info.min_delta {
+            draw_info.min_delta = delta;
+        }
 
-    for point in &mut state.points {
-        point.x /= WIDTH;
-        point.y /= HEIGHT;
+        draw_info.points.push(Point {
+            x: WIDTH,
+            y: delta, // Not the final value of y.
+            is_best_segment: false,
+        });
+        draw_info.is_live_delta_active = true;
     }
 }
 
-fn calculate_left_side_coordinates(
-    draw_info: &DrawInfo,
-    timer: &Timer,
-    total_delta: TimeSpan,
-    graph_edge: f32,
-    graph_height: f32,
-    delta: TimeSpan,
-    height_one: &mut f32,
-    width_one: &mut f32,
-    y: usize,
-) {
-    if total_delta != TimeSpan::zero() {
-        *height_one = (delta.total_milliseconds() as f32
-            - draw_info.max_delta.total_milliseconds() as f32)
-            / total_delta.total_milliseconds() as f32
-            * (graph_height - graph_edge)
-            * 2.0
-            + graph_edge;
-    } else {
-        *height_one = graph_height;
+/// Calculates the size of the chart's padding and its vertical scale factor.
+/// The padding is an area at the top/bottom that stays empty so that the graph
+/// doesn't touch the edge of the chart. This value depends on
+/// `min_`/`max_delta` because otherwise the scale factor would be huge for
+/// small graphs (graphs with only small deltas).
+fn calculate_vertical_scaling(draw_info: &mut DrawInfo) {
+    const MIN_PADDING: f32 = HEIGHT / 24.0;
+    const MAX_CONTENT_HEIGHT: f32 = HEIGHT - MIN_PADDING * 2.0;
+    const SMOOTHNESS: f32 = 0.2;
+
+    let total_delta = draw_info.max_delta - draw_info.min_delta;
+    if total_delta > 0.0 {
+        // A hyperbola works well, this looks something like f(x) = 1/(x + 2)
+        draw_info.padding_y =
+            MAX_CONTENT_HEIGHT * SMOOTHNESS / (total_delta + SMOOTHNESS * 2.0) + MIN_PADDING;
```

Comment:
Is this equivalent to what the old code was doing? Also, could you add a comment explaining what `SMOOTHNESS` is?

Reply:
Yeah, it's equivalent (aside from float rounding errors)